csrc/scheduler/normalization_utils.h

```diff
@@ -206,5 +207,10 @@ getOptionalInnerOuterPersistentBufferBatches(
     const int64_t warp_size,
     const bool ignore_register_size_limit);
 
+// Return a scheduleHeuristic based on reduction types.
+// If no reduction, returns nullptr.
+std::optional<ScheduleHeuristic> getMaybePersistentScheduleHeuristic(
```

Comment:
This seems a little confusing as it doesn't even check if a fusion should be persistent. It is just a wrapper of `getReductionType`. Instead, I'd add:

```
ScheduleHeuristic getPersistentHeuristicFor(ReductionType reduction_type) {
 switch (reduction_type) {
    case ReductionType::Inner:
      return ScheduleHeuristic::InnerPersistent;
    // OuterPersistent is not added yet, use Persistent for now.
    case ReductionType::Outer:
      return ScheduleHeuristic::Persistent;
    case ReductionType::InnerOuter:
      return ScheduleHeuristic::InnerOuterPersistent;
    default:
      NVF_ERROR(false);
  }
}
```